### PR TITLE
Ashtn RideShare 2.0

### PR DESCRIPTION
**Pull request by ashtn**
# Ride Share
Congratulations! You're submitting your assignment!

## Comprehension Questions
| Question | Answer    |
| :------------- | :------------- |
| Describe a design decision you had to make when working on this project. What options were you considering? What helped you make your final decision? | I made a design decision to keep the ratings as integers instead of converting them to be floats. As a consumer my preference is the float version of ratings, and my first instinct was to convert them. What helped me make my decision was recognizing that this project was an assignment and that it wasn't really necessary, but also that it could be tricky to test the average ratings.      |
| Describe a concept that you gained more clarity on as you worked on this assignment. |  Calling methods inside of other methods is the concept that FINALLY clicked in this project (every method has one job), mainly because y'all laid it out so simple for us *(thank you!)*, but it was something I struggled to visualize in the scrabble project.  |
| Describe a _nominal test_ that you wrote for this assignment. | A nominal test I wrote for the driver class was to ensure self.find(driver_id) returned the correct driver. Which meant I compared the return values from the method call (name, id, and vin) with what I hard coded.   |
| Describe an _edge case test_ that you wrote for this assignment. |  For the driver or rider class, in the self.all method I ensured the first and last items on the csv could be found when the method was called.   |
| How do you feel you did in writing pseudocode first, then writing the tests and then the code? | Pseudocode was very helpful in helping organize my thoughts but it was a little difficult to continue writing specs before wiring code, but I really tried this time! |

I'd like to add that my grandmother came into town this week and I ran out of time to fully complete the project (mostly adding exceptions). I plan to meet with a tutor this week, because I really struggled in that area, especially knowing when and exception (with or without a rescue) is necessary vs a puts statement. 

also had major dyslexia behavior with driver and rider. I broke my code so many times =/

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,5 @@
+
+Heya!
```
Comment (CheezItMan) [inline README.md:2]: Heya

**Comment by CheezItMan** [comment]
# Ride Share
## What We're Looking For

| Feature | Feedback    |
| :------------- | :------------- |
|  **Baseline** |   |
|   Used Git Regularly	|   Nicely done	|
|   Answer comprehension questions	|  Well done 	|
|  **Driver** |   |
|   Uses the `all` method in the `find` method |   Check  |
|   Has appropriate edge-case tests for each method in the class |   Some done, you're of course still working   |
|   Created a method that uses a method from the `Trip` object to retrieve the list of trips |  Check  |
|   Created a method that uses the internal trips list to calculate the average rating | Check |
|  **Rider** |   |
|   Uses the `all` method in the `find` method |   Check  |
|   Has appropriate edge-case tests for each method in the class | It does look like you know what to test, just haven't gotten'em tested yet.     |
|   Created a method that uses a method from the `Trip` object to retrieve the list of trips | Check |
|   Created a method that uses the internal trips method to retrieve the associated drivers | Check |
|  **Trip** |   |
|   Reads the CSV file in the `all` method | Check |
|   Has appropriate edge-case tests for each method in the class |  In progress   |
|   Created a method that uses a method from the `Driver` to retrieve the associated driver instance  |  Check  |
|   Created a method that uses a method from the `Rider` to retrieve the associated rider instance |  Check  |
|   Created a method to retrieve all trips by driver id |  Check   |
|   Created a method to retrieve all trips by rider id |    Check  |

## Summary

Overall nicely done.  I can see you ran out of time, but you were on the right track.  Let me know when you want more feedback on your tests.
